src/TimeModeling/seismic_preconditioners.jl

```diff
@@ -189,15 +225,31 @@ function find_water_bottom(m)
     return idx
 end
 
+
+function find_water_bottom(m::Array{avDT,3};eps = 1e-4) where {avDT}
```

Comment:
Making your life lot harder. Always try to reuse as much as you have

`find_water_bottom(m::Array{avDT,3};eps = 1e-4) where {avDT} =  reshape(find_water_bottom(reshape(m, :, size(m, end));eps=eps), size(m,1), size(m,2))`

same blelow

Reply:
Yes clean up find_water_bottom for 3D in the latest commit; also added 2D/3D test for find_water_bottom function